### PR TITLE
Add most of the widget tests; add more web test shards

### LINKED ISSUE #41913: Fix widget tests and enable them on CI in Web mode
We currently skip all `packages/flutter/test/widgets` tests on CI because they fail. We should fix them.

### PR DESCRIPTION
**Pull request by yjbanov**
## Description

Add most of the /test/widgets/ tests to the CI matrix and increase the number of Web test shards.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/41913

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -182,6 +182,36 @@ task:
       container:
         cpu: 4
         memory: 12G
+    - name: web_tests-linux-shard-3
+      use_compute_credits: $CIRRUS_USER_COLLABORATOR == 'true'
+      env:
+        SHARD: web_tests
+        WEB_SHARD: 2
```
Comment (ferhatb) [inline .cirrus.yml]: should have diff number

File: packages/flutter/test/widgets/html_element_view_test.dart [hunk 2]
```diff
@@ -196,7 +196,7 @@ void main() {
       // is not yet in the tree.
       await tester.pump();
 
-      final SemanticsNode semantics = tester.getSemantics(find.byType(HtmlElementView));
+      final SemanticsNode semantics = tester.getSemantics(find.byType(PlatformViewSurface));
```
Comment (ferhatb) [inline packages/flutter/test/widgets/html_element_view_test.dart]: Please add comment about id

**Review by ferhatb — APPROVED** [review]

File: packages/flutter/test/widgets/html_element_view_test.dart (hunk 2 above)
Comment (yjbanov) [reply packages/flutter/test/widgets/html_element_view_test.dart]: Done.

File: .cirrus.yml (hunk 1 above)
Comment (yjbanov) [reply .cirrus.yml]: Done.

**Pull request merged**